bundles/org.openhab.core.io.rest.sitemap/src/main/java/org/openhab/core/io/rest/sitemap/internal/SitemapResource.java

```diff
@@ -265,10 +267,50 @@ public Response getSitemapData(@Context HttpHeaders headers,
         return Response.ok(responseObject).build();
     }
 
+    /**
+     *
+     * Subscribe to a whole sitemap (all pages at once).
+     * This is not a recommended option as it incurs high traffic
+     * and might pose a high load on the server, depending on the sitemap size.
+     * No built-in openhab UI should use this functionality.
+     *
+     */
+    @GET
+    @Path("/{sitemapname: [a-zA-Z_0-9]+}/*")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(operationId = "pollDataForSitemap", summary = "Polls the data for a whole sitemap. Not recommended due to potentially high traffic.", responses = {
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = SitemapDTO.class))),
+            @ApiResponse(responseCode = "404", description = "Sitemap with requested name does not exist"),
+            @ApiResponse(responseCode = "400", description = "Invalid subscription id has been provided.") })
+    public Response getSitemapData(@Context HttpHeaders headers,
+            @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @Parameter(description = "language") @Nullable String language,
+            @PathParam("sitemapname") @Parameter(description = "sitemap name") String sitemapname,
+            @QueryParam("subscriptionid") @Parameter(description = "subscriptionid") @Nullable String subscriptionId,
+            @QueryParam("includeHidden") @Parameter(description = "include hidden widgets") boolean includeHiddenWidgets) {
+        final Locale locale = localeService.getLocale(language);
+        logger.debug("Received HTTP GET request from IP {} at '{}'", request.getRemoteAddr(), uriInfo.getPath());
+
+        if (subscriptionId != null) {
+            try {
+                subscriptions.setPageId(subscriptionId, sitemapname, null);
+            } catch (IllegalArgumentException e) {
+                return JSONResponse.createErrorResponse(Response.Status.BAD_REQUEST, e.getMessage());
+            }
+        }
+
+        boolean timeout = false;
+        if (headers.getRequestHeader("X-Atmosphere-Transport") != null) {
+            timeout = blockUnlessChangeOccursAnywhere(sitemapname);
+        }
+        SitemapDTO responseObject = getSitemapBean(sitemapname, uriInfo.getBaseUriBuilder().build(), locale,
+                includeHiddenWidgets, timeout);
```

Comment:
Why do you need timeout information in the response ?
It was not needed in the current implementation.

Reply:
Right, some of these changes might have crept in because I started development quite some time ago when the code still was like that. Same with the nullability problems. I will fix it this evening.

Reply:
I remember why the timeout field is necessary. It was meant analogous to the timeout in PageDTO, but I can just use the field timeout in the homepage of the sitemap.

Reply:
Timeout parameter removed from SitemapDTO, but still needed in getSitemapBean to set on homepage PageDTO. The parameter is necessary to tell the caller whether a change has occurred or just a timeout when long-polling for events